libs/product/src/reducers/composite-product-entities/composite-product-entities.reducer.ts

```diff
@@ -57,13 +54,24 @@ export function daffCompositeProductEntitiesReducer<T extends DaffProduct, V ext
 function buildCompositeProductAppliedOptionsEntity(product: DaffCompositeProduct): DaffCompositeProductEntity {
 	return {
 		id: product.id,
-		items: product.items.reduce((acc, item) => ({
-			...acc,
-			[item.id]: getDefaultOption(item)
-		}), {})
+		items: product.items.map(item => ({
+			code: item.id.toString(),
+			value: getDefaultOption(item),
+			qty: 1
+		}))
 	}
 }
 
+function applyItem(currentItems: DaffCompositeProductEntityItem[], appliedItemCode: string, appliedItemValue: string, quantity: number): DaffCompositeProductEntityItem[] {
+	const itemIndex = currentItems.findIndex(item => item.code === appliedItemCode);
+	currentItems[itemIndex] = {
+		code: appliedItemCode,
+		value: appliedItemValue,
+		qty: quantity
+	}
+	return currentItems;
+}
```

Comment:
This should not mutate `currentItems`.

Reply:
fixed because I'm not working with arrays anymore.